app/controllers/admin/locations_controller.rb

```diff
@@ -22,12 +22,14 @@ def edit
 
     def update
       updater = UpdateLocation.new(location: @location, user: current_user)
-      updater.update!(permitted_attributes(@location))
-      flash[:notice] = "Successfully updated #{@location.title}"
-      redirect_to admin_locations_path(location_params)
-    rescue
-      flash[:error] = "Error updating #{@location.title}"
-      redirect_to admin_locations_path(location_params)
+      @location = updater.update(permitted_attributes(@location))
+      if @location.new_record?
+        @booking_locations = policy_scope(Location.current.where(booking_location_uid: nil))
+        render :edit
+      else
+        flash[:notice] = "Successfully updated #{@location.title}"
+        redirect_to admin_location_path(@location.uid)
```

Comment:
`redirect_to` can accept `notice:` as a parameter and then you could collapse these two lines.